tests/smartswitch/platform_tests/test_platform_dpu.py

```diff
@@ -318,3 +318,30 @@ def test_system_health_summary(duthosts, dpuhosts,
         pytest_assert(result,
                       "{} health status is not ok"
                       .format(dpu_name))
+
+
+def test_data_control_mid_plane_sync(duthosts,
+                                     enum_rand_one_per_hwsku_hostname,
+                                     platform_api_conn, num_dpu_modules):  # noqa: F811
+    """
+    @summary: To verify data, control and mid planes are in sync
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    logging.info("Executing pre-test check")
+    ip_address_list, dpu_on_list, dpu_off_list = pre_test_check(
```

Comment:
suggestion : this pre_test_check can be a setup fixture which will return back data, if you plan to use in multiple testcases. **eg:** https://github.com/sonic-net/sonic-mgmt/blob/07c24ef2b4311026d8a5d484ecd27e43eaf153f1/tests/acl/test_acl.py#L267